File: src/main/java/com/example/desafio/api/models/repositorys/CarModelProjection.java
package com.example.desafio.api.models.repositorys;

// Interface serve para que o JPA consiga fazer a projeção dos dados e retornar no formato que desejo
public interface CarModelProjection {
    Long getId();
    Long getTimestampRegister();
    Long getModelId();
    String getModelName();
    Integer getYear();
    String getFuel();
    Integer getNumDoors();
    String getColor();
    Double getPriceFip();
}

File: src/main/kotlin/com/example/desafio/api/models/repositorys/CarModelProjection.kt
package com.example.desafio.api.models.repositorys

import java.math.BigDecimal

// "Interface" serve para que o JPA consiga fazer a projeção dos dados e retornar no formato que desejo
interface CarModelProjection {
    val id: Long?
    val timestampRegister: Long?
    val modelId: Long?
    val modelName: String?
    val year: Int?
    val fuel: String?
    val numDoors: Int?
    val color: String?
    val priceFip: BigDecimal?
}